myjdapi/myjdapi.py

```diff
@@ -576,26 +607,36 @@ def action(self, path, params=(), http_action="POST"):
         :param postparams: List of Params that are send in the post.
         """
         action_url = self.__action_url()
-        response = self.myjd.request_api(path, http_action, params, action_url)
+        if self.direct_connection and self.direct_connection_established is None:
+            self.direct_connect()
+        response = self.myjd.request_api(path, http_action, params, action_url, self.device_api)
         if response is None:
+            if self.direct_connection_established:
+                self.direct_connect()
+                if not self.direct_connection_established:
```

Comment:
What if there is a direct connection? Here if it fails it uses my.jdownloader but what if has a direct connection now? It doesn't use it and returns False when it could use it.